torch/jit/frontend.py

```diff
@@ -141,9 +141,11 @@ def _uses_true_division(fn):
 
 
 def get_jit_class_def(cls, self_name):
-    # Get defs for each method independently
+    # Get defs for each method within the current class independently
+    # TODO: proper overriding analysis when implementing class inheritance
     methods = inspect.getmembers(
-        cls, predicate=lambda m: inspect.ismethod(m) or inspect.isfunction(m))
+        cls, predicate=lambda m: (inspect.ismethod(m) or inspect.isfunction(m)) and m.__name__ in cls.__dict__)
```

Comment:
Why did this change?

Reply:
This is to make sure that we are only compiling the user defined functions in the module interface declaration, so that none of the methods in `nn.Module` is trying to be compiled.

Reply:
This change doesn't seem fully general to me. For example, for regular class B which inherits from class A, shouldn't A's methods be part of the interface?